### DISCUSSION TITLE
Create Content table for bookmap even with toc element in bookmap

### DISCUSSION THREAD

**Discussion by changke79** [dita-ot/dita-ot#4455] [Q&A]
Hello There, In DitaMap content table is generated based on hierarchy of the elements in a DITA map. By default, each <topicref> element in a map represents a node in the TOC.
But in Bookmap if there is no <toc> element under   <frontmatter>/<booklists>, dita-ot would not generate TOC. Is there dita-ot parameter which will allow generating TOC even without adding <toc> under <frontmatter>/<booklists>?

Thank you for your help!

**Comment by raducoravu** [comment]
As far as I know in a bookmap you need to place a &lt;toc> element otherwise no table of contents will be produced.
This XSLT stylesheet "DITA-OT/plugins/org.dita.pdf2/xsl/fo/root-processing.xsl" is the one which handles TOC processing and for plain DITA Maps it always calls the table of contents generation but for bookmaps it relies on the toc element being present.

**Reply by changke79** [reply]
Radu, Thank you!
